shhext/api.go

```diff
@@ -0,0 +1,185 @@
+package shhext
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/p2p/discover"
+	whisper "github.com/ethereum/go-ethereum/whisper/whisperv6"
+)
+
+const (
+	aesKeyLength = 32 // in bytes
+)
+
+// NewPublicAPI returns instance of the public API.
+func NewPublicAPI(handler ConfirmationHandler) *PublicAPI {
+	return &PublicAPI{
+		handler: handler,
+		tracker: map[common.Hash]bool{},
+	}
+}
+
+// PublicAPI extendes whisper public API.
+type PublicAPI struct {
+	w       *whisper.Whisper
+	handler ConfirmationHandler
+
+	mu      sync.Mutex
+	tracker map[common.Hash]bool
+}
+
+func (api *PublicAPI) track(hash common.Hash) {
```

Comment:
Sounds uncommon. The name says _"Track this hash."_ but inside the method the tracker for the hash is set to false.

Reply:
i need also to deduplicate events received by tracker somehow, that's why i have this false/true semantic

Reply:
IC. Mind adding an according comment?